File: quarkus-tmdb/runtime/src/main/java/dev/reuss/tmdb/quarkus/runtime/TmdbHealthCheck.java
package dev.reuss.tmdb.quarkus.runtime;

import dev.reuss.tmdb.TmdbClient;
import jakarta.inject.Inject;
import org.eclipse.microprofile.health.HealthCheck;
import org.eclipse.microprofile.health.HealthCheckResponse;
import org.eclipse.microprofile.health.Readiness;

/**
 * Readiness check for the TMDB SDK client.
 */
@Readiness
public class TmdbHealthCheck implements HealthCheck {

    private final TmdbClient client;

    @Inject
    public TmdbHealthCheck(TmdbClient client) {
        this.client = client;
    }

    @Override
    public HealthCheckResponse call() {
        return HealthCheckResponse.named("tmdb")
                .status(client != null)
                .build();
    }
}

File: quarkus-tmdb/runtime/src/main/kotlin/dev/reuss/tmdb/quarkus/runtime/TmdbHealthCheck.kt
package dev.reuss.tmdb.quarkus.runtime

import dev.reuss.tmdb.TmdbClient
import jakarta.inject.Inject
import org.eclipse.microprofile.health.HealthCheck
import org.eclipse.microprofile.health.HealthCheckResponse
import org.eclipse.microprofile.health.Readiness

/**
 * Readiness check that confirms the Quarkus TMDB client bean was created.
 *
 * The check deliberately performs no network request and therefore does not assert TMDB API
 * reachability or token validity.
 */
@Readiness
class TmdbHealthCheck
    @Inject
    constructor(
        @Suppress("unused")
        client: TmdbClient,
    ) : HealthCheck {
        override fun call(): HealthCheckResponse =
            HealthCheckResponse
                .named("tmdb")
                .up()
                .build()
    }
